props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,787 @@
+    }
+ }
+
+  void Loop() override {
+    PropBase::Loop();
+    PollNextAction();
+    sound_library_.Poll(wav_player);
+    if (wav_player && !wav_player->isPlaying()) wav_player.Free();
+
+    if (blaster_mode == MODE_AUTO) {
+      DoAutoFire();
+    }
+  }
+
+  void Reload() override {
+    Blaster::Reload();
+    is_jammed_ = false;
+  }
+
+  void ClipOut() override {
+    Blaster::ClipOut();
+    no_clip_ = true;
+  }
+
+  void ClipIn() override {
+    Blaster::ClipIn();
+    is_jammed_ = false;
+    no_clip_ = false;
+  }
+
+  // Pull in parent's SetPreset, but turn the blaster on.
+  void SetPreset(int preset_num, bool announce) override {
+    PropBase::SetPreset(preset_num, announce);
+     if (!SFX_poweron && !SaberBase::IsOn()) {
+      On();
+    }
+  }
+
+  void selfDestruct() override {
+    Blaster::selfDestruct();
+    SaberBase::DoEffect(EFFECT_DESTRUCT, 0);
+    SetNextActionF(NEXT_ACTION_BLOW, hybrid_font.GetCurrentEffectLength());
+  }
+
+  // Clash to unjam or Enter/Exit Volume Menu.
+  void Clash(bool stab, float strength) override {
+    PropBase::Clash(stab, strength);
+    if (is_jammed_) {
+      is_jammed_ = false;
+      SaberBase::DoEffect(EFFECT_UNJAM, 0);
+    }
+  }
+
+  // Previous, next, or first preset, depending on angle
+  void DoChangePreset() {
+    if (fusor.angle1() > M_PI / 3) {
+    // Muzzle pointing UP
+      first_preset();
+      PVLOG_NORMAL << "** Jumped to first preset\n";
+    } else if (fusor.angle1() < -M_PI / 3) {
+    // Muzzle pointing DOWN
+      previous_preset();
+      PVLOG_NORMAL << "** Previous preset\n";
+    } else {
+    // Muzzle NOT pointing UP or DOWN
+      next_preset();
+      PVLOG_NORMAL << "** Next preset\n";
+    }
+  }
+
+  void DoSpokenBatteryLevel() {
+    // Avoid weird battery readings when using USB
+    if (battery_monitor.battery() < 0.5) {
+      sound_library_.SayTheBatteryLevelIs();
+      sound_library_.SayDisabled();
+      return;
+    }
+    sound_library_.SayTheBatteryLevelIs();
+    // pointing DOWN
+    if (fusor.angle1() < -M_PI / 4) {
+      sound_library_.SayNumber(battery_monitor.battery(), SAY_DECIMAL);
+      sound_library_.SayVolts();
+      PVLOG_NORMAL << "Battery Voltage: " << battery_monitor.battery() << "\n";
+      is_speaking_ = true;
+      SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
+    } else {
+      sound_library_.SayNumber(battery_monitor.battery_percent(), SAY_WHOLE);
+      sound_library_.SayPercent();
+      PVLOG_NORMAL << "Battery Percentage: " <<battery_monitor.battery_percent() << "%\n";
+      is_speaking_ = true;
+      SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
+    }
+  }
+
+  void DoQuote() {
+    if (SFX_quote) {
+      if (GetWavPlayerPlaying(&SFX_quote)) return;  // Simple prevention of quote overlap
+      sequential_quote_ ? SFX_quote.SelectNext() : SFX_quote.Select(-1);
+      hybrid_font.PlayCommon(&SFX_quote);
+    }
+  }
+
+  void ToggleSequentialQuote() {
```

Comment:
Seems like a settings menu thing (optional)

Reply:
Maybe if there were more actions and/or buttons involved it could be, sure. But if it's ok as a simple toggle here like this, I'll choose the option to leave it.